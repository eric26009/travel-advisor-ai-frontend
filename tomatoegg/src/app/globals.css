@tailwind base;
@tailwind components;
@tailwind utilities;

* {
  margin: 0;
  padding: 0;
  background-color: rgb(248, 248, 241);
}

@layer components {
  .btn-primary {
    @apply w-[200px] lg:w-1/2 xl:w-1/2 py-2 px-4 rounded bg-black text-white font-bold border border-transparent hover:bg-white hover:text-black hover:border-black disabled:hover:bg-black disabled:hover:text-white disabled:hover:border-transparent;
  }
  .tab-inactive {
    @apply bg-gray-100 inline-block py-2 px-4 w-full border-black border rounded-tl rounded-tr;
  }
  .tab-active {
    @apply bg-gray-100 inline-block py-2 px-4 w-full border-black border rounded-tl rounded-tr border-b-[#F3F4F6];
  }
}
