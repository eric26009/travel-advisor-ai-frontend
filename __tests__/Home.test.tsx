import { render, screen } from "@testing-library/react";
import Home from "@/app/page";
import "@testing-library/jest-dom";
import { expect, test, describe } from "@jest/globals";
import userEvent from "@testing-library/user-event";

describe("Home Page", () => {
  test("should have welcome text", () => {
    const { container } = render(<Home />);
    const myElement = screen.getByText(
      "Plan your next trip using our state of the art AI travel advisor"
    );
    expect(container).toMatchSnapshot();
    expect(myElement).toBeInTheDocument();
  });

  test("should have button to take to activity explorer page", async () => {
    const user = userEvent.setup();
    render(<Home />);
    const myButtonElement = screen.getByText("Discover your next adventure");
    expect(myButtonElement).toBeInTheDocument();
    expect(myButtonElement).toHaveAttribute("href", "/explorer?type=activity");
  });
});
