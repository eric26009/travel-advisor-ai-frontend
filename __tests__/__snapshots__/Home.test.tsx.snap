// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should have welcome text 1`] = `
<div>
  <div
    class="container homepage gap-9"
  >
    <h1
      class="text-[2rem] bg-gradient-to-r from-blue-600 via-green-500 to-indigo-400 inline-block text-transparent bg-clip-text"
    >
      Plan your next trip using our state of the art AI travel advisor
    </h1>
    <a
      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
      href="/explorer?type=activity"
    >
      Discover your next adventure
    </a>
  </div>
</div>
`;
